modules/unsupported/mbstyle/src/main/java/org/geotools/mbstyle/function/MBFunction_coalesce.java

```diff
@@ -0,0 +1,60 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ * 
+ *    (C) 2018, Open Source Geospatial Foundation (OSGeo)
+ * 
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ * 
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.mbstyle.function;
+
+import static org.geotools.filter.capability.FunctionNameImpl.parameter;
+
+import org.geotools.filter.FunctionExpressionImpl;
+import org.geotools.filter.capability.FunctionNameImpl;
+import org.opengis.filter.capability.FunctionName;
+import org.opengis.filter.expression.Expression;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Evaluates each expression in turn until the first non-null value is obtained, and returns that value.
+ * Example: ["coalesce", OutputType, OutputType, ...]: OutputType
```

Comment:
Please provide a proper CQL or code example as well (i.e. how you would call it when not using MBStyle), to better illustrate how it should be used within GeoTools)